### ISSUE TITLE
Support for iconix

### ISSUE THREAD

**Issue by poperigby** [KaylorBen/nixcord#22]
Somebody made a Python script that will patch Vesktop to use the default Discord icons: https://github.com/Arcitec/iconix

It would be very useful to have this as an option.

**Comment by KaylorBen** [comment]
This feels somewhat difficult to pull off, but its certainly possible. I'll keep this open and try to work on it when I get the chance.

**Comment by joinemm** [comment]
The desktop icon can be easily patched if your installed icon theme includes a discord icon:

```nix
package = pkgs.vesktop.overrideAttrs (prev: {
  desktopItems = [
    (pkgs.makeDesktopItem {
      name = "discord";
      desktopName = "Discord";
      exec = "vesktop %U";
      icon = "discord";
      startupWMClass = "Discord";
      genericName = "Internet Messenger";
      keywords = [
        "discord"
        "vencord"
        "vesktop"
      ];
    })
  ];
});
```

**Comment by poperigby** [comment]
@joinemm that will only patch the desktop icon though, not all the in-application icons.

**Comment by Arcitec** [comment]
Actually... if @joinemm posted the actual way that Nix installs Vesktop, then your system is generating a .desktop file launcher that looks for an icon named "discord":

https://github.com/KaylorBen/nixcord/issues/22#issuecomment-2399290397

If so, all you have to do is this as the basic command:

```
./iconix -n discord
```

You also have to specify an icon.

Either the default Discord icon:

```
./iconix -n discord -d
```

Or one of my beautiful, custom icons:

```
./iconix -n discord -i icon_packs/motion_fans/silver_noir/square.png
```

**Comment by KaylorBen** [comment]
Yes, this method should work on NixOS even. Going to close for now unless there is large interest in adding options for it.
